app/src/main/java/com/github/se/polyfit/data/api/SpoonacularAPI.kt

```diff
@@ -168,4 +170,78 @@ class SpoonacularApiCaller {
+   *
+   * @param recipeId The ID of the recipe
+   * @return The response from the API
+   * @throws Exception if the response is not successful or an error occurs
+   */
+  fun getRecipeSteps(recipeId: Long): RecipeInstruction {
+    val request =
+        Request.Builder()
+            .url(API_URL + RECIPE_STEPS_ENDPOINT.format(recipeId))
+            .get()
+            .addHeader("X-RapidAPI-Key", BuildConfig.X_RapidAPI_Key)
+            .addHeader("X-RapidAPI-Host", BuildConfig.X_RapidAPI_Host)
+            .build()
+
+    try {
+      val response = client.newCall(request).execute()
+      if (!response.isSuccessful) {
+        Log.e("SpoonacularApiCaller", "Error getting recipe steps: ${response.code}")
+        throw Exception("Error getting recipe steps: ${response.code}")
+      }
+
+      val responseBody = response.body?.string() ?: ""
+      val jsonArray = JSONArray(responseBody)
+
+      // Check if the response contains any instructions
+      if (jsonArray.length() == 0) {
+        Log.e("SpoonacularApiCaller", "No instructions found for recipe: $recipeId")
+        throw Exception("No instructions found for recipe: $recipeId")
+      }
+
+      return RecipeInstructionResponseAPI.fromJson(jsonArray.getJSONObject(0).toString())
+    } catch (e: Exception) {
+      Log.e("SpoonacularApiCaller", "Error getting recipe steps", e)
+      // avoids crashing the app
+      return RecipeInstruction.faillure()
+    }
+  }
+
+  /**
+   * Get a list of recipes from a list of ingredients. This also takes care of calling other
+   * endPoints from the api to make sure the recipe is complete. For now to avoid calling the api
+   * too much, we will return a default recipe and calling the other endpoints to complete the
+   * recipe.
+   *
+   * @param ingredients The list of ingredients
+   * @return The list of recipes
+   * @throws Exception if the response is not successful or an error occurs
+   */
+  fun getCompleteRecipesFromIngredients(ingredients: List<String>): List<Recipe> {
+    //        val recipesResponse = recipeByIngredients(ingredients)
+    return try {
+
+      // done to avoid making a bunch of API calls, will need to remove later
+      val recipesResponse =
+          RecipeFromIngredientsResponseAPI(APIResponse.SUCCESS, listOf(Recipe.default()))
+      recipesResponse.recipes.forEach { recipe ->
+        val recipeInfo = getRecipeSteps(recipe.id)
+
+        // Add the steps to the recipe
+        recipe.recipeInformation.instructions = recipeInfo.steps.map { it.step }
+
+        // Add information about the ingredients, removing duplicates
+        val ingredients = recipeInfo.steps.flatMap { it.ingredients ?: emptyList() }
+        recipe.recipeInformation.ingredients = ingredients.distinctBy { it.name }
+      }
+      return recipesResponse.recipes
+    } catch (e: Exception) {
+      Log.e("SpoonacularApiCaller", "Error getting recipe from ingredients", e)
+      return emptyList()
```

Comment:
```suggestion
     recipesResponse.recipes
    } catch (e: Exception) {
      Log.e("SpoonacularApiCaller", "Error getting recipe from ingredients", e)
      emptyList()
```

Since you are returning on line 226 you don't need to return here, just have the value

Reply:
Yep good catch.